File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       unordered_map<int,int> s;
       vector<int> res;
       for(int i=0;i<nums.size();i++){
        int complement=target-nums[i];

        if(s.find(complement)!=s.end()){
            if(i<s[complement]){
            res.push_back(i);
            res.push_back(s[complement]);
            }
            else{
            res.push_back(s[complement]);
            res.push_back(i); 
            }
        }
        else{
            s[nums[i]]=i;
        }
       }
       return res;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res=nums[0]
        sub=nums[0]

        for i in range(1,len(nums)):
            n=i-1
            if sub<0:
                sub=nums[i]
            else:
                sub=sub+nums[i]
            res=max(res,sub)
        
        return res